logfire/propagate.py

```diff
@@ -70,3 +71,26 @@ def attach_context(carrier: ContextCarrier) -> Iterator[None]:
         yield
     finally:
         context.attach(old_context)
+
+
+def build_span_link(carrier: ContextCarrier) -> Link:
+    """Build a span link from a context carrier.
+
+    This is useful when you want to link a span in a different service to the current span.
+
+    ```py
+    from logfire.propagate import build_span_link, get_context
+
+    logfire_context = get_context()
+
+    ...
+
+    # later on in another thread, process or service
+    link = build_span_link(logfire_context)
+    with logfire.span('process_data', _links=[link]):
```

Comment:
```suggestion
    with logfire.span('outer span'):
        # can't use attach_context because we want to keep outer span as the parent
        with logfire.span('process_data', _links=[link]):
```